fastlane/lib/fastlane/actions/carthage.rb

```diff
@@ -11,8 +11,13 @@ def self.run(params)
 
         if command_name == "archive" && params[:frameworks].count > 0
           cmd.concat(params[:frameworks])
-        elsif ["update", "build", "bootstrap"].include?(command_name) && params[:dependencies].count > 0
-          cmd.concat(params[:dependencies])
+        elsif ["update", "build", "bootstrap"].include?(command_name)
+          if params[:dependencies].count > 0
+            cmd.concat(params[:dependencies])
+          end
+          if params[:derived_data]
+            cmd << "--derived-data #{params[:derived_data].shellescape}"
```

Comment:
I printed out all `carthage` commands:

```
Available commands:

   archive           Archives built frameworks into a zip that Carthage can use
   bootstrap         Check out and build the project's dependencies
   build             Build the project's dependencies
   checkout          Check out the project's dependencies
   copy-frameworks   In a Run Script build phase, copies each framework specified by a SCRIPT_INPUT_FILE and/or SCRIPT_INPUT_FILE_LIST environment variables into the built app bundle
   fetch             Clones or fetches a Git repository ahead of time
   help              Display general or command-specific help
   outdated          Check for compatible updates to the project's dependencies
   update            Update and rebuild the project's dependencies
   validate          Validate that the versions in Cartfile.resolved are compatible with the Cartfile requirements
   version           Display the current version of Carthage
```

and checked if only `update, build and bootstrap` support `--derived-data` parameter - indeed these are the only ones 👍 

FYI you can check that by calling `carthage help command_name` command, i.e. 

```
$ carthage help fetch                                            
Clones or fetches a Git repository ahead of time

[--color (color)]
        whether to apply color and terminal formatting (one of 'auto', 'always', or 'never')

(URL)
        the Git repository that should be cloned or fetched
```